### PR TITLE
Fixed WithdrawFromCommonPool logic and ensured compatibliity with the interface

### PR DESCRIPTION
**Pull request by vedantvasugupta**
Passed the value of the common pool to the withdraw function of the agent and created a function to handle all the withdrawals instead of implementing the logic in the runTurn function.

### CODE REVIEW

**Comment by arnavxkohli** [comment]
LGTM @vedantvasugupta , The only addition I would suggest is randomising the order in which withdrawals are made. This would ensure that withdrawals aren't skewed.

**Comment by kishan-v** [comment]
Same thoughts as [this](https://github.com/ADimoska/SOMASExtended/pull/32#discussion_r1858589481)

File: server/EnvironmentServer.go [hunk 1]
```diff
@@ -45,16 +67,8 @@ func (cs *EnvironmentServer) RunTurn(i, j int) {
...
-			if !cs.IsAgentDead(agent.GetID()) {
-				team := cs.teams[agent.GetTeamID()]
-				agentWithdrawal := agent.WithdrawFromCommonPool()
-				team.CommonPool -= agentWithdrawal
-				cs.teams[agent.GetTeamID()] = team
-				agent.SetTrueScore(agent.GetTrueScore() + agentWithdrawal)
-			}
-		}
+		cs.UpdateCommonPools()  // Update only after contributions
+		cs.ProcessWithdrawals() // Use the new withdrawal processing function
```
Comment (kishan-v) [inline server/EnvironmentServer.go:71]: Same thoughts as [this](https://github.com/ADimoska/SOMASExtended/pull/32#discussion_r1858589481)

**Review by kishan-v — CHANGES_REQUESTED** [review]

**Comment by kishan-v** [comment]
@vedantvasugupta is this PR now obsolete since #44 has been merged in?

**Comment by vedantvasugupta** [comment]
Yes I have closed this pr

**Pull request closed without merging**